### PR TITLE
SRCH-294 cache I14Y results

### PR DESCRIPTION
**Pull request by MothOnMars**
Cache I14y responses to speed up I14y searches and reduce the number of queries we send to I14y.

### CODE REVIEW

File: lib/i14y_collections.rb [hunk 1]
```diff
@@ -22,7 +26,7 @@ def self.get(handle)
   end
 
   def self.search(params)
-    response = i14y_connection.get "#{API_ENDPOINT}/search", params
-    response.body
+    response = cached_connection.get "#{API_ENDPOINT}/search", params
```
Comment (peggles2) [inline lib/i14y_collections.rb]: I think we decided to do parenthesis around method calls with arguments.

Comment (peggles2) [reply lib/i14y_collections.rb]: While your in this file can you update line 2 to be API_ENDPOINT = "/api/v1/collections".freeze?  :)

**Comment by MothOnMars** [comment]
@peggles2 , this is ready for re-review. I made some additional changes per rubocop's recommendations.

File: lib/i14y_collections.rb (hunk 1 above)
Comment (MothOnMars) [reply lib/i14y_collections.rb]: Good suggestions - updated.

Note: I froze strings in those files using the `# frozen_string_literal: true` magic comment, as that will be the default in Ruby 3.

**Comment by MothOnMars** [comment]
Closing this in favor of https://github.com/GSA/search-gov/pull/222.

**Pull request closed without merging**